packages/fluentui/react-northstar/src/components/Tree/hooks/useTreeActiveState.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import { useAutoControlled } from '@fluentui/react-bindings';
+import { TreeProps } from '../Tree';
+import { useStableProps } from './useStableProps';
+import { FlatTree } from './flattenTree';
+import { findIndex, removeItemAtIndex } from './utils';
+import * as _ from 'lodash';
+
+export interface TreeActiveState {
+  activeItemIds: string[];
+  flatTree: FlatTree;
+  toggleActive: (ids: string[], e: React.SyntheticEvent) => void;
+}
+
+export function useTreeActiveState(
+  props: Pick<TreeProps, 'defaultActiveItemIds' | 'activeItemIds' | 'exclusive'>,
+  flatTree: FlatTree,
+): TreeActiveState {
+  const initialActiveItemIds = React.useMemo(() => {
```

Comment:
```suggestion
  // We need this because we want to handle `expanded` prop on `items`, should be deprecated and removed
  const initialActiveItemIds = React.useMemo(() => {
```